zebra-consensus/src/transaction.rs

```diff
@@ -195,8 +194,8 @@ where
                         check::validate_joinsplit_sig(joinsplit_data, shielded_sighash.as_bytes())?;
                     }
 
-                    if let Some(shielded_data) = shielded_data {
-                        check::shielded_balances_match(&shielded_data, *value_balance)?;
+                    if let Some(shielded_data) = sapling_shielded_data {
+                        check::shielded_balances_match(&shielded_data, shielded_data.value_balance)?;
```

Comment:
This function need to be changed to just `shielded_balances_match(&shielded_data)` as `value_balance` is now inside.

Reply:
Go for it!